perf/proxy_service_notifier.go

```diff
@@ -0,0 +1,104 @@
+package perf
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+// PerformanceAnalysisProxyService is the interface for the Proxy Service.
+type PerformanceAnalysisProxyService interface {
+	ReportNewPerformanceDataAvailability(context.Context, model.PerformanceTestResultID) error
+}
+
+type performanceAnalysisProxyServiceClient struct {
+	user    string
+	token   string
+	baseURL string
+}
+
+// NewPerformanceAnalysisProxyService creates a new PerformanceAnalysisProxyService.
+func NewPerformanceAnalysisProxyService(options model.PerformanceAnalysisProxyServiceOptions) PerformanceAnalysisProxyService {
+	return &performanceAnalysisProxyServiceClient{user: options.User, token: options.Token, baseURL: options.BaseURL}
+}
+
+// ReportNewPerformanceDataAvailability takes a PerformanceTestResultID and tries to report its data to a PerformanceAnalysisProxyService.
```

Comment:
`// ReportNewPerformanceDataAvailability reports the given ID to the underlying performance analysis proxy service.`